libfreerdp/codec/rfx_neon.c

```diff
@@ -225,6 +225,296 @@ static void rfx_dwt_2d_decode_NEON(INT16* buffer, INT16* dwt_buffer)
+				h_n_m = vsetq_lane_s16(first, h_n_m, 0);
+			}
+			else if (n == 24)
+			{
+				h_n = vsetq_lane_s16(0, h_n, 7);
+			}
+
+			int16x8_t tmp_n = vaddq_s16(h_n, h_n_m);
+			tmp_n = vaddq_s16(tmp_n, vdupq_n_s16(1));
+			tmp_n = vshrq_n_s16(tmp_n, 1);
+			int16x8_t dst_n = vsubq_s16(l_n, tmp_n);
+			vst1q_s16(l_ptr, dst_n);
+			l_ptr += 8;
+			h_ptr += 8;
+		}
+		if (n < 32)
+		{
+			*l_ptr -= *(h_ptr - 1);
+		}
+
+		l_ptr -= batchSize;
+		h_ptr -= batchSize;
+
+		/* Odd coefficients */
+		for (n = 0; n < batchSize; n += 8)
+		{
+			// dst[2n + 1] = (h[n] << 1) + ((dst[2n] + dst[2n + 2]) >> 1);
+			int16x8_t h_n = vld1q_s16(h_ptr);
+			h_n = vshlq_n_s16(h_n, 1);
+			int16x8x2_t dst_n;
+			dst_n.val[0] = vld1q_s16(l_ptr);
+			int16x8_t dst_n_p = vld1q_s16(l_ptr + 1);
+
+			if (n == 24)
+			{
+				h_n = vsetq_lane_s16(0, h_n, 7);
+			}
+
+			dst_n.val[1] = vaddq_s16(dst_n_p, dst_n.val[0]);
+			dst_n.val[1] = vshrq_n_s16(dst_n.val[1], 1);
+			dst_n.val[1] = vaddq_s16(dst_n.val[1], h_n);
+			vst2q_s16(dst_ptr, dst_n);
+			l_ptr += 8;
+			h_ptr += 8;
+			dst_ptr += 16;
+		}
+		if (n == 32)
+		{
+			h_ptr -= 1;
+			l_ptr += 1;
+		}
+		else
+		{
+			*dst_ptr = *l_ptr;
+			l_ptr += 1;
+			dst_ptr += 1;
+		}
+	}
+}
+
+static INLINE void __attribute__((__gnu_inline__, __always_inline__, __artificial__))
+rfx_idwt_extrapolate_vert_neon(const INT16* restrict pLowBand, size_t nLowStep,
+                               const INT16* restrict pHighBand, size_t nHighStep,
+                               INT16* restrict pDstBand, size_t nDstStep, size_t nLowCount,
+                               size_t nHighCount, size_t nDstCount)
+{
+	int x, n;
+	INT16* l_ptr = pLowBand;
+	INT16* h_ptr = pHighBand;
+	INT16* dst_ptr = pDstBand;
+	int batchSize = (nDstCount >> 3) << 3;
+	int forceBandSize = (nLowCount + nHighCount) >> 1;
+
+	/* Even coefficients */
+	for (n = 0; n < forceBandSize; n++)
+	{
+		for (x = 0; x < batchSize; x += 8)
+		{
+			// dst[2n] = l[n] - ((h[n-1] + h[n] + 1) >> 1);
+			int16x8_t l_n = vld1q_s16(l_ptr);
+			int16x8_t h_n = vld1q_s16((n == 31) ? (h_ptr - nHighStep) : h_ptr);
+			int16x8_t tmp_n = vaddq_s16(h_n, vdupq_n_s16(1));
+
+			if (n == 0)
+			{
+				tmp_n = vaddq_s16(tmp_n, h_n);
+			}
+			else if (n < 31)
+			{
+				int16x8_t h_n_m = vld1q_s16((h_ptr - nHighStep));
+				tmp_n = vaddq_s16(tmp_n, h_n_m);
+			}
+
+			tmp_n = vshrq_n_s16(tmp_n, 1);
+			int16x8_t dst_n = vsubq_s16(l_n, tmp_n);
+			vst1q_s16(dst_ptr, dst_n);
+			l_ptr += 8;
+			h_ptr += 8;
+			dst_ptr += 8;
+		}
+
+		if (nDstCount > batchSize)
+		{
+			int16_t h_n = (n == 31) ? *(h_ptr - nHighStep) : *h_ptr;
+			int16_t tmp_n = h_n + 1;
+			if (n == 0)
+			{
+				tmp_n += h_n;
+			}
+			else if (n < 31)
+			{
+				tmp_n += *(h_ptr - nHighStep);
+			}
+			tmp_n >>= 1;
+			*dst_ptr = *l_ptr - tmp_n;
+			l_ptr += 1;
+			h_ptr += 1;
+			dst_ptr += 1;
+		}
+
+		dst_ptr += nDstStep;
+	}
+
+	if (forceBandSize < 32)
+	{
+		for (x = 0; x < batchSize; x += 8)
```

Comment:
we´re using modern C now, you can declare the variable in the loop like `for (size x = 0; x < batchSize; x++)`
also ensure the type matches (you have `int x` and `size_t` for `batchSize` here)